packages/core/src/extensions/SideMenu/SideMenuPlugin.ts

```diff
@@ -283,22 +286,42 @@ export class SideMenuView<
       this.pmView.dom.firstChild! as HTMLElement
     ).getBoundingClientRect().x;
 
-    document.body.addEventListener("drop", this.onDrop, true);
-    document.body.addEventListener("dragover", this.onDragOver);
+    // document.body.addEventListener("drop", this.onDrop, true);
+    this.pmView.root.addEventListener(
+      "drop",
+      this.onDrop as EventListener,
```

Comment:
Can we make this work without the cast?

Reply:
Without the cast no, there is another way where we explicitly type it with the type of the handler. Like so:

this.pmView.root.addEventListener(
      "drop",
      (event: Event) => this.onDrop(event as DragEvent),
      true
    );
    
Let me know the one you prefer to keep as standard and I can do the necessary updates!

Reply:
Can't we also conditionally add the listeners only if `view.root instanceof Document`? Since shadow roots don't support listeners anyway